### PR TITLE
adding exec function

### PR DESCRIPTION
**Pull request by iezell**
Adding the exec function. Calls bash for darwin/linux and cmd for windows. can call run function directly for the cli for now, but likely change with a wrapper function to call extra functions with extra functionality i.e. parallel runs.

### CODE REVIEW

File: internal/exec/exec.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+	"syscall"
+)
+
+type Execute struct {
+	StdOut   string
+	StdErr   string
+	ExitCode int
+}
+
+func isExecutable(path string) bool {
```
Comment (mja8020) [inline internal/exec/exec.go:21]: Do we need to bother checking for this?  All good if its just be thorough

File: internal/exec/exec.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+	execPath, err := exec.LookPath(executable)
+	if err != nil {
+		return result, fmt.Errorf("executable not found or not in PATH: %s", executable)
+	}
+	if !isExecutable(execPath) {
+		return result, fmt.Errorf("executable is not permitted to run: %s", execPath)
+	}
+
+	// Set up command
+	cmd := exec.Command(execPath, strings.Join(args, " "))
```
Comment (mja8020) [inline internal/exec/exec.go:42]: Dont think you need to join here, command takes 0+ args.  Variadic (is that the term lol)?

File: internal/exec/exec.go [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
```
Comment (mja8020) [inline internal/exec/exec.go:43]: Maybe use CommandContext so we can cancel? I think @terrabyteian  would know

Comment (iezell) [reply internal/exec/exec.go:31]: swapped it over to accept context now. just added a basic one for the actual cobra-cli call. dont really anticipate that being the file function. its more or less to test out the functionality, but i think we should maybe support a base shell call for debugging or anything like that.

File: internal/exec/exec.go (hunk 2 above)
Comment (iezell) [reply internal/exec/exec.go]: didn't realize args... was a thing.

File: internal/exec/exec.go (hunk 1 above)
Comment (iezell) [reply internal/exec/exec.go]: removed

**Review by mja8020 — APPROVED** [review]

**Pull request merged**